extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
 	//2. Add date range input field
 
 	const PAGINATION_SIZE = 50;
+	const REQUEST_SIZE = 200;
 	const DAYS_RANGE = 15;
 	const HTTP_STATUS_OK = 200;
-	const NO_USER_MATCH_ERROR = "Provided username did not match any user";
+	const NO_USER_MATCH_ERROR = 'discussionslog-no-user-match-error';
 
 	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
 
+	private $logger;
+	private $users;
+
 	public function __construct() {
 		parent::__construct( 'DiscussionsLog', '', false );
+		$this->users = [ ];
+		$this->logger = Wikia\Logger\WikiaLogger::instance();
 	}
 
 	public function index() {
 
-		if ( !$this->checkAccess() ) {
-			throw new \PermissionsError( 'specialdiscussionslog' );
-		}
+		$this->checkAccess();
 
 		$this->setHeaders();
+		$this->response->addAsset( '/extensions/wikia/SpecialDiscussionsLog/css/DiscussionsLog_Forms.scss' );
 
 		$this->wg->Out->setPageTitle( wfMessage( 'discussionslog-pagetitle' )->escaped() );
 
-		$userName = $this->wg->request->getVal( 'username', null );
+		list( $userName, $ipAddress ) = $this->readAndValidateInput();
 
-		$this->inputForm = $this->sendSelfRequest( 'inputForm', [ 'username' => $userName ] );
-		if ( !empty( $userName ) ) {
-			$this->userLog = $this->sendSelfRequest( 'userLog', [ 'username' => $userName ] );
+		if ( $userName ) {
+			$searchQuery = new UserQuery( $userName );
+		} else if ( $ipAddress ) {
+			$searchQuery = new IpAddressQuery( $ipAddress );
+		} else {
+			$searchQuery = null;
```

Comment:
Oh, nothing is wrong there! I just had my mind set on minimizing the number of lines :)